### ISSUE TITLE
Snapshot apply fails on constraint even just changing sort

### ISSUE THREAD

**Issue by u12206050** [directus/directus#18523]
### Describe the Bug

For a long time now I have been manually fixing these issues directly in the database, but since there is still no fix, I guess I just have to start with an issue :)

So basically I have a snapshot with the following difference:
``` 
  - Update organizations.entity
    - Set sort to 5
  - Update organizations.tpts
    - Set hidden to true
    - Set sort to 4
    - Set group to Relationships
```

This is after moving the `tpts` field into the `Relationships` group right before the existing `entity` field.

On running `snapshot apply` against another environment I get the following error:
```
[08:31:20.665] ERROR: Failed to update field "organizations.entity"
[08:31:20.770] ERROR: alter table `organizations` modify `entity` varchar(36) - ER_FK_COLUMN_CANNOT_CHANGE: Cannot change column 'entity': used in a foreign key constraint 'organizations_entity_foreign'
    err: {
      "type": "Error",
      "message": "alter table `organizations` modify `entity` varchar(36) - ER_FK_COLUMN_CANNOT_CHANGE: Cannot change column 'entity': used in a foreign key constraint 'organizations_entity_foreign'",
      "stack":
          Error: ER_FK_COLUMN_CANNOT_CHANGE: Cannot change column 'entity': used in a foreign key constraint 'organizations_entity_foreign'
              at Sequence._packetToError (~/cms/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
              at Query.ErrorPacket (~/cms/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
              at Protocol._parsePacket (~/cms/node_modules/mysql/lib/protocol/Protocol.js:291:23)
              at Parser._parsePacket (~/cms/node_modules/mysql/lib/protocol/Parser.js:433:10)
              at Parser.write (~/cms/node_modules/mysql/lib/protocol/Parser.js:43:10)
              at Protocol.write (~/cms/node_modules/mysql/lib/protocol/Protocol.js:38:16)
              at Socket.<anonymous> (~/cms/node_modules/mysql/lib/Connection.js:88:28)
              at Socket.<anonymous> (~/cms/node_modules/mysql/lib/Connection.js:526:10)
              at Socket.emit (node:events:513:28)
              at addChunk (node:internal/streams/readable:324:12)
              --------------------
              at Protocol._enqueue (~/cms/node_modules/mysql/lib/protocol/Protocol.js:144:48)
              at Connection.query (~/cms/node_modules/mysql/lib/Connection.js:198:25)
              at ~/cms/node_modules/directus/node_modules/knex/lib/dialects/mysql/index.js:132:18
              at new Promise (<anonymous>)
              at Client_MySQL._query (~/cms/node_modules/directus/node_modules/knex/lib/dialects/mysql/index.js:126:12)
              at executeQuery (~/cms/node_modules/directus/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
              at Client_MySQL.query (~/cms/node_modules/directus/node_modules/knex/lib/client.js:146:12)
              at ~/cms/node_modules/directus/node_modules/knex/lib/execution/transaction.js:363:24
              at new Promise (<anonymous>)
              at trxClient.query (~/cms/node_modules/directus/node_modules/knex/lib/execution/transaction.js:358:12)
      "code": "ER_FK_COLUMN_CANNOT_CHANGE",
      "errno": 1832,
      "sqlMessage": "Cannot change column 'entity': used in a foreign key constraint 'organizations_entity_foreign'",
      "sqlState": "HY000",
      "index": 0,
      "sql": "alter table `organizations` modify `entity` varchar(36)"
    }
```

It does not make sense why on earth it is trying to alter the entity field in this way if all it needs to be doing is updating the sort value in the `directus_fields` table???

### To Reproduce

Create a relationship field(1) (m2o) on collection A, and place it in a group (not sure if this has anything to do with it).
Create an alias field(2) on collection A that maps to another relationship (doesn't matter)

Now first take a snapshot and apply to another environment.

Then move the alias field(2) into the same group as the first field(1) and before it such that it will have to update sort.

Take a snapshot and attempt to apply to the other environment.

### Directus Version

9.24.0

### Hosting Strategy

Self-Hosted (Custom)

**Comment by foby** [comment]
Same here. What we do as a hacky workaround is to check the diff and update fields that cause errors on apply manually in the `directus_fields` table. Hope this issue is tackled soon, as it breaks our deployment.

**Comment by ubeytd** [comment]
Same here,  we had an "image" field related to files, we changed it to string but the change is not applied because the constraint is not dropped.  It's breaking deployment right now, @foby  can you elaborate more on your hacky solution?

**Comment by u12206050** [comment]
Hi, yes can confirm this seems to be fixed when testing it on `10.1.0`
Thanks so much.

**Comment by paescuj** [comment]
> Hi, yes can confirm this seems to be fixed when testing it on `10.1.0` Thanks so much.

Great! Thank you very much for the confirmation!
